zephyr: CMSIS header makes it unable to build for non-arm arch

Recently there was a change (#1785) that includes `cmis_header.h` without any guards making the zephyr build fail for other architectures such as Xtensa:

```
mcuboot/boot/zephyr/main.c:31:10: fatal error: cmsis_core.h: No such file or directory
   31 | #include <cmsis_core.h>
      |          ^~~~~~~~~~~~~~
```

--- a/boot/zephyr/main.c
+++ b/boot/zephyr/main.c
@@ -27,7 +27,9 @@
 #include <soc.h>
 #include <zephyr/linker/linker-defs.h>
 
+#if defined(CONFIG_ARM)
 #include <cmsis_core.h>
+#endif
 
 #include "target.h"
 

